### ISSUE TITLE
Authenticating dialog doesn't go away on Android 4.x devices

### ISSUE THREAD

**Issue by toshGitHubIrv** [box/box-android-sdk#260]
Hi,

On devices previously affected by #256 , it seems that the Authenticating spinner is not dismissed. Looking into it, I think two different spinners are generated by `OAuthActivity.showDialogWhileWaitingForAuthenticationAPICall` and the reference to the first is overridden by the second, which is later dismissed -- but the first remains even after the webview is finished loading.

**Comment by doncung** [comment]
Thanks. I've made a PR that should address that. I've tested on a 4.4 device, but let me know if you are still seeing issues after this.
https://github.com/box/box-android-sdk/pull/261

**Comment by toshGitHubIrv** [comment]
Checked out the fork and it solves the issue, thanks.

Something else I noticed is sometimes, on these devices, after losing focus on either the username or password EditView, both EditViews stop receiving keypresses. Each keypress generates the following warnings:

```
W/IInputConnectionWrapper: beginBatchEdit on inactive InputConnection
W/IInputConnectionWrapper: getTextBeforeCursor on inactive InputConnection
W/IInputConnectionWrapper: getTextAfterCursor on inactive InputConnection
W/IInputConnectionWrapper: getSelectedText on inactive InputConnection
W/IInputConnectionWrapper: endBatchEdit on inactive InputConnection
```

Not sure if you can do anything about that -- I only managed to recreate this on these same old devices.

**Comment by doncung** [comment]
The username and password are handled in a webview so that the server can handle the more unusual login cases. So it's unlikely we can do anything about that directly. I assume it's something in the webkit implementation trying to figure out if the user is trying to select something inside the webview vs moving the cursor.

**Comment by doncung** [comment]
Released 4.0.7 which has these fixes in it.

**Comment by toshGitHubIrv** [comment]
Thanks for the help!
